src/mempool/verify_tx.rs

```diff
@@ -135,6 +137,24 @@ impl super::Mempool {
                     self.lanes
                         .store_data_proposal(&crypto, &lane_id, data_proposal)?;
                 self.send_vote(self.get_lane_operator(&lane_id), hash, size)?;
+
+                let mut podas_to_process = Vec::new();
+
+                if let Some(lane) = self.inner.buffered_podas.get_mut(&lane_id) {
+                    lane.retain(|x| {
+                        if x.dp_hash == dp_hash {
+                            podas_to_process.push(x.clone());
+                            false
+                        } else {
+                            true
+                        }
```

Comment:
```suggestion
    podas_to_process.extend(
        lane.drain_filter(|x| x.dp_hash == dp_hash)
    );
```
J'aurais tenté un truc comme ça

Reply:
j'ai extract une fonction